feat: Add support for different color prediction lines

Erin879 opened the pull request:
This pull request lets users modify the colors of each individual OpenAPS prediction line (iob, cob, zt, and uam).

It adds 4 new environment variables to configure these colors: `OPENAPS_PRED_IOB`, `OPENAPS_PRED_COB`, `OPENAPS_PRED_ZT`, and `OPENAPS_PRED_UAM` which take a hex color code as input. The default values of these are using the current color that is applied to all prediction lines (`#ff00ff`).

I decided to stick with the same color for all of them as the default, but I am open to opinions on whether they should have new defaults or if the current approach is best.

unsoluble commented:
This is a great idea.

PieterGit commented on README.md:
```diff
@@ -478,6 +478,11 @@ To learn more about the Nightscout API, visit https://YOUR-SITE.com/api-docs.htm
+  * `OPENAPS_PREB_IOB` (`#ff00ff`) - The color to use for IOB prediction lines
+  * `OPENAPS_PREB_COB` (`#ff00ff`) - The color to use for COB prediction lines
+  * `OPENAPS_PREB_ZT` (`#ff00ff`) - The color to use for ZT prediction lines
+  * `OPENAPS_PREB_UAM` (`#ff00ff`) - The color to use for UAM prediction lines
```
Where does `PREB` stand for? Perhaps a better naming would be `OPENAPS_PRED_IOB_COLOR`

sulkaharo replied:
Reading the code, this is a typo :)

PieterGit commented on lib/plugins/openaps.js:
```diff
@@ -39,6 +39,10 @@ function init (ctx) {
       , warn: sbx.extendedSettings.warn ? sbx.extendedSettings.warn : 30
       , urgent: sbx.extendedSettings.urgent ? sbx.extendedSettings.urgent : 60
       , enableAlerts: sbx.extendedSettings.enableAlerts
+      , predIOB: sbx.extendedSettings.predIob ? sbx.extendedSettings.predIob : '#ff00ff'
```
Naming: perhaps `predIOBColor`  instead of `predIOB` to clearly indicate it's just about the color

PieterGit commented:
@Erin879 @unsoluble Nice PR. I justed switched from oref0 to loop and I must say that I really like the fact that there is only one prediction line with loop. That's much easier to understand. 

I wouldn't mind to have desperate colors by default, but I think those should be carefully chosen. Perhaps @scottleibrand or @danamlewis have opinions on that.

Perhaps we can tune the naming a bit, but otherwise the PR looks good. If it's been tested by another developer. I think it's good to add a feature like this. Adding for the 0.13 milestone.

sulkaharo commented:
@Erin879 this is a great thing, can you fix the above issues quick and I'll probably push this to the the next .3 release that's about to go out to address other issues

Erin879 commented:
@sulkaharo @PieterGit I've gone ahead and fixed those issues in the latest commit

sulkaharo commented:
Hey one more thing actually - should we color those by default? I'd basically just get the hex codes from AndroidAPS and apply the same colors in Nightscout by default

Erin879 commented:
@sulkaharo Using the colors from AndroidAPS sounds like a great idea - would you like me to include that change in this PR?

sulkaharo commented:
Yes please! Also actually note there can be two cob prediction lines, depending on oref version in use.

```
<color name="iob">#1e88e5</color>
<color name="cob">#FFFB8C00</color>
<color name="acob">#80FB8C00</color>
<color name="uam">#c9bd60</color>
<color name="zt">#00d2d2</color>
```

Erin879 commented:
I've applied those colors as defaults, but I've been unable to test the acob preditions - but I can't imagine any reasons for why it'd break. I've also added info to the readme about how to format the color variables, and changed the default color for the 'Values' predictions to the old purple to be safe, as I'm unsure on what exactly that does.

sulkaharo commented:
(Unrelated: everyone add `co-pr = !sh -c 'git fetch origin pull/$1/head:pr/$1 && git checkout pr/$1' -` to your .gitconfig, so you can say `git co-pr 4792` to see this in all of it's glory easily

PieterGit commented:
Can't test it, but LGTM and approved by @sulkaharo . 
Since we messed up the 0.12.x semver rules, we don't bother about semver rules for now.

danamlewis commented:
I would make the color change an option and not on by default - it will
confuse a lot of people.

On Wed, Jul 24, 2019 at 10:24 AM PieterGit <notifications@github.com> wrote:

> Can't test it, but LGTM and approved by @sulkaharo
> <https://github.com/sulkaharo> .
> Since we messed up the 0.12.x semver rules, we don't bother about semver
> rules for now.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nightscout/cgm-remote-monitor/pull/4792?email_source=notifications&email_token=ABY7JBMQALXZX3TMVWTNZKTQBCF3XA5CNFSM4IGKS4HKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2XBA3A#issuecomment-514723948>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABY7JBMO54IIK4KN2LSZABTQBCF3XANCNFSM4IGKS4HA>
> .
>
-- 
Dana Lewis | http://OpenAPS.org | http://DIYPS.org <http://diyps.org> |
@DanaMLewis <http://www.twitter.com/danamlewis> |
http://www.linkedin.com/in/danalewis
"Doing something for someone else is more important than anything you would
do for yourself."
